crates/app/src/lib.rs

```diff
@@ -207,6 +207,22 @@ impl<'a, L> App<'a, L> {
             .map(|locked| AppTrigger { app: self, locked })
     }
 
+    /// Returns the trigger metadata for a specific trigger type.
+    pub fn get_trigger_metadata<'this, T: Deserialize<'this> + Default>(
```

Comment:
Unfortunately we need to maintain backwards-compatibility in the locked app manifest, so we can't just change things around like this.

Reply:
Note that I'm not changing the format here. the .lock file has something like (for a single trigger, multiple triggers are supported but don't write out the single trigger item):
```
{
  "spin_lock_version": 0,
  "metadata": {
    "authors": [
      "itowlson <ivan.towlson@fermyon.com>"
    ],
    "name": "cron-sample",
    "origin": "file:///C:/projects/spintest/spin.toml",
    "triggers": {
      "http": {
        "base": "/shop"
      }
    },
    "trigger": {
      "type": "http": 
      "base": "/shop"
    },
    "version": "0.1.0"
  },
```

Reply:
Ah sorry, I added the `"triggers"` key as part of the migration to Spin 2.0 but we still need to be able to read the (redundant) old `"trigger"` keys for backward-compat. I need to figure out how to document all of this somewhere...

Reply:
There's potentially also an issue where plugins which are interested in trigger types may not know about the `metadata/triggers` key, but rely on `metadata/trigger`.  (Although my main worry was the Cloud plugin, and that looks directly at the trigger entries rather than the metadata.)

This may be addressable by continuing to populate `metadata/trigger` if there is a single type, and setting it to something like `type: composite` if there are multiple types?  That way, existing "check trigger type" code will still work, and the error if it encounters a composite app should be semi-meaningful.

Reply:
Fixed this with a fallback to look at either.